packages/webapp/src/components/Form/Unit/useUnit.jsx

```diff
@@ -103,85 +106,84 @@ const useUnit = ({
+    if (displayUnit && (measure === 'time' || convert().describe(displayUnit)?.system === system)) {
+      return {
+        ...values,
+        displayUnit,
+        displayValue: value && roundToTwoDecimal(convert(value).from(databaseUnit).to(displayUnit)),
+      };
+    }
 
-    return to && convert().describe(to)?.system === system
-      ? {
-          displayUnit: to,
-          displayValue: defaultValue && roundToTwoDecimal(convert(value).from(databaseUnit).to(to)),
-          options,
-          databaseUnit,
-          defaultIsSelectDisabled,
-          measure,
-          reactSelectWidth,
-          dividerWidth,
-          onKeyDown,
-        }
-      : {
-          ...getDefaultUnit(unitType, value, system, databaseUnit),
-          options,
-          databaseUnit,
-          defaultIsSelectDisabled,
-          measure,
-          reactSelectWidth,
-          dividerWidth,
-          onKeyDown,
-        };
+    // system generated/create/unit system unmatch: let the function determine a unit based on the guidance
+    return { ...values, ...getDefaultUnit(unitType, value, system, databaseUnit) };
   }, []);
 
   const hookFormUnitOption = hookFromWatch(displayUnitName);
-  const hookFormUnit = databaseUnit;
-  useEffect(() => {
-    if (typeof hookFormUnitOption === 'string' && getUnitOptionMap()[hookFormUnitOption]) {
-      hookFormSetValue(displayUnitName, getUnitOptionMap()[hookFormUnitOption]);
-    }
-  }, []);
-  useEffect(() => {
-    if (hookFormUnit && convert().describe(hookFormUnit)?.system !== system && measure !== 'time') {
-      hookFormSetValue(displayUnitName, getUnitOptionMap()[displayUnit]);
-    }
-  }, [hookFormUnit]);
+  const hookFormUnit = hookFormUnitOption?.value;
 
+  // 2. Set defaultDisplayUnit
   useEffect(() => {
-    !hookFormGetValue(displayUnitName) &&
-      hookFormSetValue(displayUnitName, getUnitOptionMap()[displayUnit]);
+    const unit = getUnitOptionMap()[defaultDisplayUnit];
+    hookFormSetValue(displayUnitName, unit);
   }, []);
 
-  const [visibleInputValue, setVisibleInputValue] = useState(displayValue);
+  // 3. set default visibleInputValue
+  const [visibleInputValue, setVisibleInputValue] = useState(defaultVisibleInputValue);
+
   const hookFormValue = hookFromWatch(name, defaultValue);
 
   useEffect(() => {
-    hookFormSetHiddenValue(hookFormValue, { shouldValidate: true, shouldDirty: false });
-  }, []);
+    if (!hookFormUnit || noValue(hookFormValue)) {
+      return;
+    }
 
-  useEffect(() => {
-    if (hookFormUnit && hookFormValue !== undefined) {
+    // if autoConversion is true, update visible value. if false, update hidden value
+    if (autoConversion) {
       setVisibleInputValue(
         roundToTwoDecimal(convert(hookFormValue).from(databaseUnit).to(hookFormUnit)),
       );
       //Trigger validation
       (hookFormValue === 0 || hookFormValue > 0) && hookFormSetHiddenValue(hookFormValue);
```

Comment:
why it excludes negative values?

Reply:
https://github.com/LiteFarmOrg/LiteFarm/pull/2079
This is one reason pull requests should be squashed per ticket.
It's very difficult to track changes in git tree